lib/core/src/sync/client.rs

```diff
@@ -18,58 +26,97 @@ pub(crate) trait SyncerClient: Send + Sync {
 }
 
 pub(crate) struct BreezSyncerClient {
-    inner: Mutex<Option<ProtoSyncerClient<tonic::transport::Channel>>>,
+    grpc_channel: Mutex<Option<Channel>>,
     api_key: Option<String>,
 }
 
 impl BreezSyncerClient {
     pub(crate) fn new(api_key: Option<String>) -> Self {
         Self {
-            inner: Default::default(),
+            grpc_channel: Mutex::new(None),
             api_key,
         }
     }
+
+    fn create_endpoint(server_url: &str) -> Result<Endpoint> {
+        Ok(Endpoint::from_shared(server_url.to_string())?
+            .http2_keep_alive_interval(Duration::new(5, 0))
+            .tcp_keepalive(Some(Duration::from_secs(5)))
+            .keep_alive_timeout(Duration::from_secs(5))
+            .keep_alive_while_idle(true)
+            .tls_config(ClientTlsConfig::new().with_enabled_roots())?)
+    }
+
+    fn api_key_metadata(&self) -> Result<Option<MetadataValue<Ascii>>, Error> {
+        match &self.api_key {
+            Some(key) => Ok(Some(format!("Bearer {key}").parse().map_err(
+                |e: InvalidMetadataValue| {
+                    anyhow!(format!(
+                        "(Breez: {:?}) Failed parse API key: {e}",
+                        self.api_key
+                    ))
+                },
+            )?)),
+            _ => Ok(None),
+        }
+    }
 }
 
 impl BreezSyncerClient {
-    fn set_api_key<T>(&self, req: T) -> Result<tonic::Request<T>> {
-        let mut req = tonic::Request::new(req);
-        if let Some(api_key) = &self.api_key {
-            let metadata = req.metadata_mut();
-            metadata.insert("authorization", format!("Bearer {}", api_key).parse()?);
-        }
-        Ok(req)
+    async fn get_client(
+        &self,
+    ) -> Result<ProtoSyncerClient<InterceptedService<Channel, ApiKeyInterceptor>>, Error> {
+        let Some(channel) = self.grpc_channel.lock().await.clone() else {
+            return Err(anyhow!("Cannot run `set_record`: client not connected"));
```

Comment:
Maybe a clearer error message?
```suggestion
            return Err(anyhow!("Cannot get client: gRPC not available"));
```

Reply:
Fixed